client/my-sites/stats/activity-log/index.jsx

```diff
@@ -234,17 +234,17 @@ class ActivityLog extends Component {
 		const logsGroupedByDay = map(
 			groupBy(
 				logsForMonth,
-				log => applySiteOffset( moment.utc( log.ts_utc ) ).format( 'LL' )
+				log => applySiteOffset( moment.utc( log.ts_utc ) ).startOf( 'day' ).valueOf()
 			),
-			( daily_logs, day ) => (
+			( daily_logs, tsDayStart ) => (
 				<ActivityLogDay
 					allowRestore={ !! isPressable }
 					isRewindActive={ isRewindActive }
-					key={ day }
+					key={ tsDayStart }
 					logs={ daily_logs }
 					requestRestore={ this.handleRequestRestore }
 					siteId={ siteId }
-					day={ day }
+					tsDayStart={ +tsDayStart }
```

Comment:
Timestamps _are_ utc:

```js
const ts = 3600
// valueOf returns timestamp, unaffected by tz, offeset
moment.utc( ts ).valueOf() // 3600
moment.utc( ts ).tz('Asia/Bangkok').valueOf() // 3600
moment.utc( ts ).utcOffset(12).valueOf() // 3600

// Formatting is affected
moment.utc( ts ).format('LLLL') "Thursday, January 1, 1970 12:00 AM"
moment.utc( ts ).tz('Asia/Bangkok').format('LLLL') // "Thursday, January 1, 1970 7:00 AM"
moment.utc( ts ).utcOffset(12).format('LLLL') // "Thursday, January 1, 1970 12:00 PM"
```

Passing _real_ timestamps should save us the trouble of even thinking about timezones.